File: backend/app/schemas/content_moderation.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class FlaggedSpan(BaseModel):
    """模型标出的敏感文本片段。"""

    field: str
    text: str
    start: int
    end: int
    category: str
    confidence: float

    model_config = ConfigDict(from_attributes=True)


class ContentModerationRecordResponse(BaseModel):
    """单条内容审核记录响应。"""

    id: int
    task_id: str
    trace_id: str | None
    target_type: str
    target_id: int
    target_version: str
    trigger_action: str
    status: str
    risk_level: str
    categories: list[str] | None
    original_snapshot: dict[str, Any]
    moderation_result: dict[str, Any] | None
    action_plan: dict[str, Any] | None
    action_result: dict[str, Any] | None
    model_name: str | None
    error_type: str | None
    error_message: str | None
    created_by_user_id: int | None
    created_at: datetime
    started_at: datetime | None
    finished_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class ContentModerationRecordListResponse(BaseModel):
    """内容审核记录分页列表响应。"""

    items: list[ContentModerationRecordResponse]
    total: int

    model_config = ConfigDict(from_attributes=True)


class ContentModerationStatsResponse(BaseModel):
    """内容审核统计概览响应。"""

    total: int
    today_count: int
    failed_count: int
    blocked_count: int
    avg_latency_ms: int | None

    model_config = ConfigDict(from_attributes=True)


class ContentModerationRetryResponse(BaseModel):
    """审核重试操作响应。"""

    id: int
    task_id: str
    status: str
    message: str

    model_config = ConfigDict(from_attributes=True)


class ModerationAgentOutput(BaseModel):
    """审核 Agent 结构化输出（用于解析 LLM 返回的 JSON）。"""

    verdict: str
    risk_level: str
    categories: list[str]
    reason: str
    flagged_spans: list[FlaggedSpan]
    suggested_action: str

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/contentModeration.ts
export type ModerationStatus =
  | 'pending'
  | 'running'
  | 'passed'
  | 'blocked'
  | 'action_failed'
  | 'review_failed'
  | 'stale'

export type ModerationTargetType =
  | 'user'
  | 'blog_post'
  | 'forum_zone'
  | 'forum_post'
  | 'forum_reply'
  | 'comment'

export type RiskLevel = 'none' | 'low' | 'medium' | 'high'

export interface FlaggedSpan {
  field: string
  text: string
  start: number
  end: number
  category: string
  confidence: number
}

export interface ContentModerationRecord {
  id: number
  task_id: string
  trace_id: string | null
  target_type: ModerationTargetType
  target_id: number
  target_version: string
  trigger_action: string
  status: ModerationStatus
  risk_level: RiskLevel
  categories: string[] | null
  original_snapshot: Record<string, unknown>
  moderation_result: Record<string, unknown> | null
  action_plan: Record<string, unknown> | null
  action_result: Record<string, unknown> | null
  model_name: string | null
  error_type: string | null
  error_message: string | null
  created_by_user_id: number | null
  created_at: string
  started_at: string | null
  finished_at: string | null
}

export interface ContentModerationRecordListResponse {
  items: ContentModerationRecord[]
  total: number
}

export interface ContentModerationStatsResponse {
  total: number
  today_count: number
  failed_count: number
  blocked_count: number
  avg_latency_ms: number | null
}

export interface ContentModerationRetryResponse {
  id: number
  task_id: string
  status: string
  message: string
}

export interface ContentModerationListParams {
  skip?: number
  limit?: number
  status?: ModerationStatus
  target_type?: ModerationTargetType
  risk_level?: RiskLevel
  user_id?: number
  start_time?: string
  end_time?: string
}
